src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOResearchPurposeSupport.scala

```diff
@@ -5,16 +5,62 @@ import org.broadinstitute.dsde.firecloud.model.DataUse.{DiseaseOntologyNodeId, R
 import org.broadinstitute.dsde.firecloud.model.Ontology.{TermParent, TermResource}
 import org.broadinstitute.dsde.firecloud.service.DataUseRestrictionSupport
 import org.broadinstitute.dsde.rawls.model.AttributeName
-import org.elasticsearch.index.query.BoolQueryBuilder
+import org.elasticsearch.index.query.{BoolQueryBuilder, TermQueryBuilder}
 import org.elasticsearch.index.query.QueryBuilders.{boolQuery, termQuery}
 
 import scala.concurrent.{Await, ExecutionContext, Future}
 
 trait ElasticSearchDAOResearchPurposeSupport extends DataUseRestrictionSupport with LazyLogging {
 
-  val durRoot = AttributeName.toDelimitedName(structuredUseRestrictionAttributeName)
+  def researchPurposeFilters(rp: ResearchPurpose, ontologyDAO: OntologyDAO, attributePrefix: Option[String])(implicit ec: ExecutionContext): BoolQueryBuilder = {
 
-  def researchPurposeFilters(rp: ResearchPurpose, ontologyDAO: OntologyDAO)(implicit ec: ExecutionContext): BoolQueryBuilder = {
+    val durRoot: String = attributePrefix.getOrElse("") + structuredUseRestrictionName
+
+    def generateDiseaseMatchLogic(rp: ResearchPurpose, ontologyDAO: OntologyDAO)(implicit ec: ExecutionContext): Option[BoolQueryBuilder] = {
```

Comment:
All of these functions, along with `durRoot`, moved inside `researchPurposeFilters` unchanged so they could have access to the `attributePrefix` parameter without having to pass it around.